packages/recommendations-vdom/src/RelatedProducts.tsx

```diff
@@ -0,0 +1,60 @@
+/** @jsx createElement */
+import { GetRelatedProductsProps } from '@algolia/recommendations-core';
+
+import { createDefaultChildrenComponent } from './DefaultChildren';
+import { createDefaultFallbackComponent } from './DefaultFallback';
+import { createDefaultHeaderComponent } from './DefaultHeader';
+import { createListViewComponent } from './ListView';
+import {
+  RecommendationsComponentProps,
+  RecommendationTranslations,
+  Renderer,
+} from './types';
+
+export type RelatedProductsProps<TObject> = GetRelatedProductsProps<TObject> &
+  RecommendationsComponentProps<TObject>;
+
+export function createRelatedProductsComponent({
+  createElement,
+  Fragment,
+}: Renderer) {
+  return function RelatedProducts<TObject>(
+    props: RelatedProductsProps<TObject>
+  ) {
+    const translations: Required<RecommendationTranslations> = {
+      title: 'Related products',
+      sliderLabel: 'Related products',
+      ...props.translations,
+    };
+    const classNames = props.classNames ?? {};
+
+    const children =
+      props.children ??
+      createDefaultChildrenComponent({ createElement, Fragment });
+    const Fallback =
+      props.fallbackComponent ?? createDefaultFallbackComponent();
+    const Header =
+      props.headerComponent ??
+      createDefaultHeaderComponent({ createElement, Fragment });
+    const ViewComponent =
+      props.view ?? createListViewComponent({ createElement, Fragment });
+    const View = (viewProps: unknown) => (
+      <ViewComponent
+        classNames={classNames}
+        itemComponent={props.itemComponent}
+        items={props.items}
+        translations={translations}
+        {...viewProps}
+      />
+    );
+
+    return children({
```

Comment:
should this use the children-as-a-function pattern still? not sure if that translates to Vue efficiently

Reply:
if children is a component, all those arguments could be props, not sure what the other options are

Reply:
Having children as a function makes the API powerful for custom use cases, but the fact that it might not work with Vue is a good point. We'll need to test it before public release then. I guess converting that to a slot would work in the Vue API?

Reply:
most likely, I didn't think of slots, so that should indeed solve the issue